lib/salesforce_sync/resource/action.rb

```diff
@@ -52,6 +75,21 @@ def client
       def sf_resource
         @sf_resource ||= sf_class.new(resource_id)
       end
+
+      def upsert_record
+        salesforce_id = client.upsert!(sf_class.sf_type, sf_class.resource_id_name, sf_resource.all_prepared_fields)
+        sf_resource.store_salesforce_id(salesforce_id)
+        sf_resource.after_upsert
+      end
+
+      def update_record(data)
+        client.update!("#{SF_USER_QUERY_TYPE}", Id: "#{data.Id}", Seedrs_User_ID__c: "#{sf_resource.resource.id}")
```

Comment:
On my local machine, the job failed at this line. It complained that `data` doesn't have a method Id as `data` it is a hash

Reply:
I believe the select converts the result to hash
```
def self.select(query)
  collection = new.send("client").query(query)
  return [] if collection.size == 0

  collection.map do |record|
    record.to_hash.deep_symbolize_keys
  end
end
```

Reply:
hhmmmm yeah. Originally I wasn't using the `select` method, I was using the query method of the client directly but then used `select` as it was there. I'll fix it